### DISCUSSION TITLE
Eagerly materialize a chain / DAG of assets

### DISCUSSION THREAD

**Discussion by christeefy** [dagster-io/dagster#26255] [Q&A]
Assuming I have a chain of assets (`asset1` -> `asset2` -> `asset3`) with the following intentions:
- `asset1` should be triggered via cron
- Ultimately, `asset3` (and transitively `asset2`) should be triggered eagerly whenever `asset1` materialized

Using the Declarative Automation API, the implementation would like this:
```python
from dagster import asset, AutomationCondition


@asset(automation_condition=AutomationCondition.on_cron("..."))
def asset1():
    ...

@asset(automation_condition=AutomationCondition.eager())
def asset2(asset1):
    ...

@asset(automation_condition=AutomationCondition.eager())
def asset3(asset2):
    ...
```

In the current iteration of the API (v1.9.3), `asset2`'s `AutomationCondition.eager()` needs to be specified. Without it, `asset3` wouldn't eagerly materialize. 

The issue is that this doesn't scale with many assets — data practitioners would have to review the Global Asset Graph (for complicated DAGs) to identify non-root assets and update their automation condition accordingly. Non-root assets added going forward also have to be remembered to be added with the automation condition, lest they break the "chain of eager conditions". 

Is there a clean, maintainable way to set the automation condition for various assets within a group? For a target (leaf) asset, is there a way to specify eager automation conditions for it and all upstream dependencies somehow?

**Comment by v1gnesh** [comment]
I could be wrong, but is it an option to set on_cron on asset3, and add deps=asset2 to it, and set asset2 with deps=asset1?

**Comment by ion-elgreco** [comment]
You can add a default condition to your automation sensor, this will be applied to all assets that don't have a condition set yet.

Here is also a useful example: https://github.com/dagster-io/dagster/discussions/22811#discussioncomment-11489765

You could for example do this after load_assets_from_modules
